src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -34,162 +44,541 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            return ParseDicDictionary(text);
+        }
+
+        private static CodeAnalysisDictionary ParseXmlDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromXml);
+
+        private static CodeAnalysisDictionary ParseDicDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromDic);
+
+        private static string RemovePrefixIfPresent(string prefix, string name)
+            => name.StartsWith(prefix, StringComparison.Ordinal) ? name.Substring(1) : name;
+
+        private static IEnumerable<Diagnostic> GetUnmeaningfulIdentifierDiagnostics(ISymbol symbol, string symbolName)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Assembly:
+                    yield return Diagnostic.Create(AssemblyMoreMeaningfulNameRule, Location.None, symbolName);
+                    break;
+
+                case SymbolKind.Namespace:
+                    yield return Diagnostic.Create(NamespaceMoreMeaningfulNameRule, symbol.Locations.First(), symbolName);
+                    break;
+
+                case SymbolKind.NamedType:
+                    foreach (var location in symbol.Locations)
+                    {
+                        yield return Diagnostic.Create(TypeMoreMeaningfulNameRule, location, symbolName);
+                    }
+
+                    break;
+
+                case SymbolKind.Method:
+                case SymbolKind.Property:
+                case SymbolKind.Event:
+                case SymbolKind.Field:
+                    yield return Diagnostic.Create(MemberMoreMeaningfulNameRule, symbol.Locations.First(), symbolName);
+                    break;
+
+                case SymbolKind.Parameter:
+                    yield return symbol.ContainingType.TypeKind == TypeKind.Delegate
+                        ? Diagnostic.Create(DelegateParameterMoreMeaningfulNameRule, symbol.Locations.First(), symbol.ContainingType.ToDisplayString(), symbolName)
+                        : Diagnostic.Create(MemberParameterMoreMeaningfulNameRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), symbolName);
+                    break;
+
+                case SymbolKind.TypeParameter:
+                    yield return symbol.ContainingSymbol.Kind == SymbolKind.Method
+                        ? Diagnostic.Create(MethodTypeParameterMoreMeaningfulNameRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), symbol.Name)
+                        : Diagnostic.Create(TypeTypeParameterMoreMeaningfulNameRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), symbol.Name);
+                    break;
+
+                default:
+                    throw new NotImplementedException($"Unknown SymbolKind: {symbol.Kind}");
+            }
+        }
+
+        private static IEnumerable<Diagnostic> GetMisspelledWordDiagnostics(ISymbol symbol, string misspelledWord)
```

Comment:
```suggestion
        private static Diagnostic GetMisspelledWordDiagnostic(ISymbol symbol, string misspelledWord)
```

Reply:
Actually the method can be changed to `private static void ReportMisspelledWordDiagnostic(ISymbol symbol, string misspelledWord, Action<Diagnostic> addDiagnostic)`

Reply:
Kept it "Get" but reported it sooner